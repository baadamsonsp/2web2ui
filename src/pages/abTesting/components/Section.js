import React, { Component } from 'react';
import { Grid } from '@sparkpost/matchbox';
import styles from './Section.module.scss';

const Left = ({ children }) => (
  <Grid.Column xs={12} lg={4}>
    <div className={styles.Left}>{children}</div>
  </Grid.Column>
);
const Right = ({ children }) => (
  <Grid.Column xs={12} lg={8}>
    <div className={styles.Right}>{children}</div>
  </Grid.Column>
);

export default class Section extends Component {
  static Left = Left;
  static Right = Right;

  render() {
    const { title, children } = this.props;

    return (
      <div className={styles.Section}>
        <hr className={styles.Hr}/>
        <h3>{title}</h3>
        <Grid>
          {children}
        </Grid>
      </div>
    );
  }
}
