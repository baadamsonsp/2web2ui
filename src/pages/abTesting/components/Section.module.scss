@import '~@sparkpost/matchbox/src/styles/config.scss';

.Right {
  margin-top: spacing();

  @media screen and (min-width: breakpoint(medium)) {
    margin-top: rem(-42);
  }
}
