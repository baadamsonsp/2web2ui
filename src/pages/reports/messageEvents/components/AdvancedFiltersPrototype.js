/* eslint-disable */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

import { Grid, Panel, TextField, Modal, Button, Tag, Checkbox, Tooltip } from '@sparkpost/matchbox';
import DateFilter from 'src/pages/reports/components/DateFilter';
// import { FilterDropdown, TextFieldWrapper } from 'src/components';

const tagWrapper = {
  marginRight: 6,
  marginLeft: 6
};

const boxWrapper = {
  display: 'inline-block',
  width: '25%'
};
class AdvancedFiltersPrototype extends Component {
  state = {
    modalOpen: false
  }

  render() {
    return (
      <React.Fragment>
        <Panel>
          <Panel.Section>
            <Grid>
              <Grid.Column xs={5}>
                <DateFilter />
              </Grid.Column>
              <Grid.Column xs={5}>
                <TextField placeholder='Search by recipient'/>
              </Grid.Column>
              <Grid.Column xs={2}>
                <Button
                  fullWidth
                  onClick={() => this.setState({ modalOpen: true })}>
                  More Filters
                </Button>
              </Grid.Column>
            </Grid>
          </Panel.Section>
          <Panel.Section>
            <small>Filters:</small>
            <Tag style={tagWrapper} onRemove={() => (console.log('remove'))}>
              Event: Injection
            </Tag>
            <Tag style={tagWrapper} onRemove={() => (console.log('remove'))}>
              Event: Bounce
            </Tag>
            <Tag style={tagWrapper} onRemove={() => (console.log('remove'))}>
              Recipient: 123@hoolimail.com
            </Tag>
          </Panel.Section>
        </Panel>

        <Modal open={this.state.modalOpen}>
          <Panel title='Advanced Filters'>
            <Panel.Section>
              <Checkbox.Group label='Event Type'>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Bounce' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Out of Band' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Initial Open' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Generation Failure' /></Tooltip></div>

                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Delivery' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Policy Rejection' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Open' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Generation Rejection' /></Tooltip></div>

                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Injection' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Delay' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Click' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='List Unsubscribe' /></Tooltip></div>
                <div style={boxWrapper}/>
                <div style={boxWrapper}/>

                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Link Unsubscribe' /></Tooltip></div>
                <div style={boxWrapper}><Tooltip content='🤷‍♂️' dark><Checkbox label='Spam Complaint' /></Tooltip></div>
              </Checkbox.Group>
            </Panel.Section>
            <Panel.Section>
              <Grid>
                <Grid.Column xs={6}>
                  <TextField label='Subaccount' placeholder='this should be typeahead'/>
                  <TextField label='Message ID'/>
                </Grid.Column>
                <Grid.Column xs={6}>
                  <TextField label='Template ID'/>
                  <TextField label='Campaign'/>
                </Grid.Column>
              </Grid>
            </Panel.Section>
            <Panel.Section>
              <Button primary onClick={() => this.setState({ modalOpen: false })}>Apply</Button>
              <Button style={{ marginLeft: 6 }} onClick={() => this.setState({ modalOpen: false })}>Reset</Button>
              <Button style={{ float: 'right' }} onClick={() => this.setState({ modalOpen: false })}>Cancel</Button>
            </Panel.Section>
          </Panel>
        </Modal>
      </React.Fragment>
    );
  }
}

const formName = 'msgEventsAdvancedFilter';
const formOptions = { form: formName };

const mapStateToProps = ({ reportOptions }) => ({ reportOptions });

export default connect(
  mapStateToProps
)(reduxForm(formOptions)(AdvancedFiltersPrototype));


{ /* <FilterDropdown
  formName='msgEventsBasicFilter'
  namespace='type'
  displayValue='Event Type'
  options={[
    { name: 'bounce', content: 'Bounce' },
    { name: 'delivery', content: 'Delivery' },
    { name: 'injection', content: 'Injection' },
    { name: 'injection', content: 'Injection' },
    { name: 'bounce', content: 'Bounce' },
    { name: 'delivery', content: 'Delivery' },
    { name: 'injection', content: 'Injection' },
    { name: 'injection', content: 'Injection' },
    { name: 'bounce', content: 'Bounce' },
    { name: 'delivery', content: 'Delivery' },
    { name: 'injection', content: 'Injection' },
    { name: 'injection', content: 'Injection' }
  ]}
/> */ }
