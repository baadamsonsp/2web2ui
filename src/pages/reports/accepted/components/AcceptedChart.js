import React, { Component } from 'react';
import _ from 'lodash';
import { Grid, Panel } from '@sparkpost/matchbox';
import { Loading, PieChart } from 'src/components';
import { generateColors } from 'src/helpers/color';
import styles from './AcceptedChart.module.scss';
import { formatPercent } from 'src/helpers/units';
import { safeRate } from 'src/helpers/math';

// Chart color palette generated from:
const primaryColor = '#8CCA3A';

export default class AcceptedChart extends Component {
  state = {
    hoveredItem: null,
    active: null
  };

  /**
   * Handles mouse over event for LegendItems and BounceChart
   * @param  {Object} e - Recharts synthetic event - behavior mimiced from Legend
   * @param  {string} hoverSet - 'primary' | 'secondary'
   */
  handleMouseOver = (e, hoverSet) => {
    const { attempts } = this.props;
    const { active } = this.state;
    const { name, count } = e;

    let dataSet = attempts;

    if (active) {
      dataSet = active.children;
    }

    const hoveredItem = {
      name,
      count,
      index: _.findIndex(dataSet, { name }),
      dataSet: hoverSet
    };

    this.setState({ hoveredItem });
  }

  handleMouseOut = () => {
    this.setState({ hoveredItem: null });
  }

  handleClick = ({ name, children, count }) => {
    if (children) {
      this.setState({
        active: { name, children, count },
        hoveredItem: null
      });
    }
  }

  handleBreadcrumb = () => {
    this.setState({ active: null });
  }

  getLabelProps = () => {
    const { aggregates } = this.props;
    const { hoveredItem } = this.state;

    return hoveredItem
      ? { name: hoveredItem.name, value: formatPercent(safeRate(hoveredItem.count, aggregates.count_accepted)) }
      : { name: 'Accepted Rate', value: formatPercent(safeRate(aggregates.count_accepted, aggregates.count_sent)) };
  }

  getLegendHeaderData = () => {
    const { aggregates } = this.props;
    const { active } = this.state;

    // Header with breadcrumb & active data
    if (active) {
      return [
        { name: 'Accepted', breadcrumb: true, onClick: this.handleBreadcrumb, count: aggregates.count_accepted },
        { name: 'Sent', count: aggregates.count_sent },
        { name: active.name, count: active.count }
      ];
    }

    // Default header
    return [
      { name: 'Accepted', count: aggregates.count_accepted },
      { name: 'Sent', count: aggregates.count_sent }
    ];
  }

  // Gets primary and secondary data for BounceChart & Legend
  getData = () => {
    const { attempts } = this.props;
    const { active } = this.state;

    let data = attempts;

    if (active) {
      data = active.children;
    }

    return {
      primaryData: generateColors(data, { baseColor: primaryColor, saturate: 0, rotate: -40 })
    };
  }

  render() {
    const { loading } = this.props;

    if (loading) {
      return <Panel title='Accepted Rates' sectioned className={styles.LoadingPanel}><Loading /></Panel>;
    }

    return (
      <Panel title='Accepted Rates' sectioned>
        <Grid>
          <Grid.Column xs={12} lg={5}>
            <div className={styles.ChartWrapper}>
              <PieChart.Chart
                {...this.getData()}
                hoveredItem={this.state.hoveredItem}
                onMouseOver={this.handleMouseOver}
                onMouseOut={this.handleMouseOut}
                onClick={this.handleClick} />
              <PieChart.ActiveLabel {...this.getLabelProps()}/>
            </div>
          </Grid.Column>
          <Grid.Column xs={12} lg={7}>
            <PieChart.Legend
              headerData={this.getLegendHeaderData()}
              {...this.getData()}
              hoveredItem={this.state.hoveredItem}
              onMouseOver={this.handleMouseOver}
              onMouseOut={this.handleMouseOut}
              onClick={this.handleClick} />
          </Grid.Column>
        </Grid>
      </Panel>
    );
  }
}
