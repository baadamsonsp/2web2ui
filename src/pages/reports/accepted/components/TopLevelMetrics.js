import React from 'react';
import { Grid } from '@sparkpost/matchbox';
import { Size, Duration } from 'src/components';
import { MetricCard, MetricsSummary } from '../../components';
import { safeRate } from 'src/helpers/math';
import _ from 'lodash';

/**
 * Presentational component for the accepted report metric cards
 */
const TopLevelMetrics = ({ filters, metrics, aggregates }) => {
  const {
    count_targeted,
    count_accepted,
    count_sent,
    avg_delivery_time_first,
    avg_delivery_time_subsequent,
    avg_msg_size
  } = aggregates;

  const latencyFirstCard = (
    <MetricCard
      value={<Duration value={avg_delivery_time_first}/>}
      tooltipContent={_.find(metrics, { key: 'avg_delivery_time_first' }).description}
      label='Avg Latency (First)' />
  );

  const latencySubCard = (
    <MetricCard
      value={<Duration value={avg_delivery_time_subsequent}/>}
      tooltipContent={_.find(metrics, { key: 'avg_delivery_time_subsequent' }).description}
      label='Avg Latency (Subsequent)' />
  );

  const sizeCard = (
    <MetricCard
      value={<Size value={avg_msg_size}/>}
      tooltipContent={_.find(metrics, { key: 'avg_msg_size' }).description}
      label='Avg Message Size' />
  );

  return (
    <div>
      <MetricsSummary
        rateValue={safeRate(count_accepted, count_sent)}
        rateTitle='Accepted Rate'
        secondaryMessage={`${count_targeted.toLocaleString()} messages were targeted.`}
        {...filters} >
        <strong>{count_accepted.toLocaleString()}</strong> of <strong>{count_sent.toLocaleString()}</strong> sent messages were accepted
      </MetricsSummary>
      <Grid>
        <Grid.Column xs={12} md={4}>{latencyFirstCard}</Grid.Column>
        <Grid.Column xs={12} md={4}>{latencySubCard}</Grid.Column>
        <Grid.Column xs={12} md={4}>{sizeCard}</Grid.Column>
      </Grid>
    </div>
  );
};

export default TopLevelMetrics;
