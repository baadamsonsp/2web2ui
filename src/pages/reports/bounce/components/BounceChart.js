import React, { Component } from 'react';
import _ from 'lodash';
import { Grid, Panel, Tabs } from '@sparkpost/matchbox';
import { Loading, PieChart } from 'src/components';
import { generateColors } from 'src/helpers/color';
import styles from './BounceChart.module.scss';
import { formatPercent } from 'src/helpers/units';
import { safeRate } from 'src/helpers/math';

// Chart color palette generated from:
const primaryColor = '#DB2F3D';
const secondaryColor = '#37aadc';

const fakeData = [
  { name: 'Admin Bounce', count: 249 },
  { name: 'Smart Send Suppression', count: 84 }
];

export default class BounceChart extends Component {
  state = {
    hoveredItem: null,
    active: null,
    tab: 0
  };

  /**
   * Handles mouse over event for LegendItems and BounceChart
   * @param  {Object} e - Recharts synthetic event - behavior mimiced from Legend
   * @param  {string} hoverSet - 'primary' | 'secondary'
   */
  handleMouseOver = (e, hoverSet) => {
    const { categories, types } = this.props;
    const { active, tab } = this.state;
    const { name, count } = e;

    let dataSet = hoverSet === 'primary' ? categories : types;

    if (active) {
      dataSet = active.children;
    }

    if (tab === 1) {
      dataSet = fakeData;
    }

    const hoveredItem = {
      name,
      count,
      index: _.findIndex(dataSet, { name }),
      dataSet: hoverSet
    };

    this.setState({ hoveredItem });
  }

  handleMouseOut = () => {
    this.setState({ hoveredItem: null });
  }

  handleClick = ({ name, children, count }) => {
    if (children) {
      this.setState({
        active: { name, children, count },
        hoveredItem: null
      });
    }
  }

  handleBreadcrumb = () => {
    this.setState({ active: null });
  }

  getLabelProps = () => {
    const { aggregates } = this.props;
    const { hoveredItem, tab } = this.state;

    let defaultLabel = { name: 'Bounce Rate', value: formatPercent(safeRate(aggregates.countBounce, aggregates.countSent)) };

    if (tab === 1) {
      defaultLabel = { name: 'Bounce Rate', value: formatPercent(safeRate(aggregates.countAdminBounce, aggregates.countTargeted)) };
    }

    return hoveredItem
      ? { name: hoveredItem.name, value: formatPercent(safeRate(hoveredItem.count, tab === 1 ? aggregates.countBounce : aggregates.countAdminBounce)) }
      : defaultLabel;
  }

  getLegendHeaderData = () => {
    const { aggregates } = this.props;
    const { active, tab } = this.state;

    // Header with breadcrumb & active data
    if (active) {
      return [
        { name: 'Bounces', breadcrumb: true, onClick: this.handleBreadcrumb, count: aggregates.countBounce },
        { name: 'Sent', count: aggregates.countSent },
        { name: active.name, count: active.count }
      ];
    }

    if (tab === 1) {
      return [
        { name: 'Admin Bounces', count: aggregates.countAdminBounce },
        { name: 'Targeted', count: aggregates.countTargeted }
      ];
    }

    // Default header
    return [
      { name: 'Bounces', count: aggregates.countBounce },
      { name: 'Sent', count: aggregates.countSent }
    ];
  }

  // Gets primary and secondary data for BounceChart & Legend
  getData = () => {
    const { categories, types } = this.props;
    const { active, tab } = this.state;

    let primaryData = categories;
    let secondaryData = types;

    if (active) {
      primaryData = active.children;
      secondaryData = null;
    }

    if (tab === 1) {
      primaryData = fakeData;
      secondaryData = null;
    }

    return {
      primaryData: generateColors(primaryData, { baseColor: primaryColor, rotate: 80, saturate: 0.06 }),
      secondaryData: secondaryData && generateColors(secondaryData, { baseColor: secondaryColor })
    };
  }

  handleTab = (index) => {
    this.setState({ tab: index });
  }

  render() {
    const { loading } = this.props;

    if (loading) {
      return <Panel title='Bounce Rates' sectioned className={styles.LoadingPanel}><Loading /></Panel>;
    }

    return (
      <React.Fragment>
        <Tabs tabs={[{ content: 'Bounces', onClick: () => this.handleTab(0) }, { content: 'Admin Bounces', onClick: () => this.handleTab(1) }]} selected={this.state.tab} />
        <Panel sectioned>
          <Grid>
            <Grid.Column xs={12} lg={5}>
              <div className={styles.ChartWrapper}>
                <PieChart.Chart
                  {...this.getData()}
                  hoveredItem={this.state.hoveredItem}
                  onMouseOver={this.handleMouseOver}
                  onMouseOut={this.handleMouseOut}
                  onClick={this.handleClick} />
                <PieChart.ActiveLabel {...this.getLabelProps()}/>
              </div>
            </Grid.Column>
            <Grid.Column xs={12} lg={7}>
              <PieChart.Legend
                headerData={this.getLegendHeaderData()}
                {...this.getData()}
                hoveredItem={this.state.hoveredItem}
                onMouseOver={this.handleMouseOver}
                onMouseOut={this.handleMouseOut}
                onClick={this.handleClick} />
            </Grid.Column>
          </Grid>
        </Panel>
      </React.Fragment>
    );
  }
}
