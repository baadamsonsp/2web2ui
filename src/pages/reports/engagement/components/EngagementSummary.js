import React from 'react';
import { connect } from 'react-redux';
import { Grid, Panel } from '@sparkpost/matchbox';

import { PanelLoading } from 'src/components';
import { Percent } from 'src/components/formatters';
import { formatDateTime, relativeDateOptionsIndexed } from 'src/helpers/date';
import { formatFullNumber } from 'src/helpers/units';
import { safeRate } from 'src/helpers/math';
import styles from './EngagementSummary.module.scss';

export function EngagementSummary({ accepted = 0, clicks = 0, filters, loading, opens = 0, sent = 0 }) {
  if (loading) {
    return <PanelLoading minHeight="115px" />;
  }

  // sent should always be the largest number, so if it is zero all others should be zero
  if (sent === 0) {
    return null;
  }

  const timeRange = filters.relativeRange === 'custom'
    ? <span>from <strong>{formatDateTime(filters.from)}</strong> to <strong>{formatDateTime(filters.to)}</strong></span>
    : <span>in the <strong>{relativeDateOptionsIndexed[filters.relativeRange].toLowerCase()}</strong></span>;

  return (
    <Panel className={styles.EngagementSummary}>
      <Grid>
        <Grid.Column xs={12} md={3} xl={2}>
          <h1>< Percent value={safeRate(opens, accepted)} /></h1>
          <h6>Unique Open Rate</h6>
        </Grid.Column>
        <Grid.Column xs={12} md={3} xl={2}>
          <h1>< Percent value={safeRate(clicks, accepted)} /></h1>
          <h6>Unique Click Rate</h6>
        </Grid.Column>
        <Grid.Column xs={12} md={6} xl={8}>
          <p>
            Of <strong>{formatFullNumber(sent)}</strong> sent
            recipients, <strong>{formatFullNumber(accepted)}</strong> messages were
            accepted, <strong>{formatFullNumber(opens)}</strong> were opened
            and <strong>{formatFullNumber(clicks)}</strong> were clicked {timeRange}.
          </p>
        </Grid.Column>
      </Grid>
    </Panel>
  );
}

const mapStateToProps = ({ reportOptions: filters }) => ({ filters });
export default connect(mapStateToProps)(EngagementSummary);
