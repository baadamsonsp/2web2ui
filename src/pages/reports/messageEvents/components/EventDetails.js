import React, { Component } from 'react';

import { Panel, Tooltip } from '@sparkpost/matchbox';
import { LabelledValue, CopyField } from 'src/components';

import _ from 'lodash';

class EventDetails extends Component {
  static defaultProps = {
    details: {}
  }

  renderRow = ({ value, label, key }) => {
    // Renders value in a read only textfield
    if (typeof value === 'object') {
      if (key === 'geo_ip') {
        //define the coordinates
        const long = value.longitude;
        const lat = value.latitude;

        //image with the coordinates applied to the url
        const dynmap = `https://www.google.com/maps/embed/v1/place?key=AIzaSyCPEyzwYdKNq-lsqOn7fcYivTWpuZirWNI&q=${lat},${long}`;
        return (
          <LabelledValue key={key} label={label}>
            <CopyField hideCopy value={JSON.stringify(value)} />
            <iframe src= {dynmap}/>
          </LabelledValue>

        );
      } else {
        return (
          <LabelledValue key={key} label={label}>
            <CopyField hideCopy value={JSON.stringify(value)} />
          </LabelledValue>
        );
      }
    }

    return <LabelledValue key={key} label={label} value={value} />;
  }

  // Renders key-value rows
  renderDetails = (detailsToRender) => {
    const { documentation } = this.props;

    return _.keys(detailsToRender).map((key) => {

      // Gets tooltip content from documentation store
      const tooltipContent = _.get(documentation, [detailsToRender.type, key]);

      const row = {
        key,
        label: tooltipContent ? <Tooltip dark content={tooltipContent}>{key}</Tooltip> : key,
        value: detailsToRender[key]
      };

      return this.renderRow(row);
    });
  }

  render() {
    // Remove formattedDate as its only used for the history table
    const detailsToRender = _.omit(this.props.details, 'formattedDate');

    return (
      <Panel title='Event Details'>
        <Panel.Section>
          {this.renderDetails(detailsToRender)}
        </Panel.Section>
        <Panel.Section>
          <LabelledValue label='Raw Json'>
            <CopyField value={JSON.stringify(detailsToRender)}/>
          </LabelledValue>
        </Panel.Section>
      </Panel>
    );
  }
}

export default EventDetails;
