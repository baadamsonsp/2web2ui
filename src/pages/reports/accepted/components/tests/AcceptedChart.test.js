import React from 'react';
import AcceptedChart from '../AcceptedChart';
import { PieChart } from 'src/components';

import { shallow } from 'enzyme';

describe('Accepted ChartGroup: ', () => {

  const props = {
    attempts: [
      { name: 'first attempt', count: 5 },
      { name: '2 or more attempts', count: 1, children: [{ name: '2-4', count: 2 }, { name: '5+', count: 3 }]}
    ],
    aggregates: {
      count_targeted: 100,
      count_accepted: 50,
      count_sent: 80
    }
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AcceptedChart {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle mouse in and out', () => {
    wrapper.instance().handleMouseOver(props.attempts[0]);
    wrapper.update();
    expect(wrapper.instance().state).toMatchSnapshot();
    expect(wrapper.find(PieChart.ActiveLabel)).toMatchSnapshot();

    wrapper.instance().handleMouseOut();
    wrapper.update();
    expect(wrapper.find(PieChart.ActiveLabel)).toMatchSnapshot();
    expect(wrapper.instance().state).toMatchSnapshot();
  });

  it('should render label when hovering over a child', () => {
    wrapper.instance().handleClick(props.attempts[1]);
    wrapper.instance().handleMouseOver({ name: '2-4', count: 2 });
    wrapper.update();
    expect(wrapper.find(PieChart.ActiveLabel)).toMatchSnapshot();
  });

  it('should render correctly after click', () => {
    wrapper.instance().handleClick(props.attempts[1]);
    wrapper.update();
    expect(wrapper.find(PieChart.Legend)).toMatchSnapshot();
    expect(wrapper.find(PieChart.Chart)).toMatchSnapshot();
  });

  it('should not do anything if clicking on an item without children', () => {
    wrapper.instance().handleClick(props.attempts[0]);
    expect(wrapper.instance().state).toMatchSnapshot();
  });

  it('should handle breadcrumb', () => {
    wrapper.instance().handleClick(props.attempts[1]);
    expect(wrapper.instance().state).toMatchSnapshot();
    wrapper.instance().handleBreadcrumb();
    expect(wrapper.instance().state).toMatchSnapshot();
  });
});
