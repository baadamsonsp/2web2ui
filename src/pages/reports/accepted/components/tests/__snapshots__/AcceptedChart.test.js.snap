// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accepted ChartGroup:  should handle breadcrumb 1`] = `
Object {
  "active": Object {
    "children": Array [
      Object {
        "count": 2,
        "name": "2-4",
      },
      Object {
        "count": 3,
        "name": "5+",
      },
    ],
    "count": 1,
    "name": "2 or more attempts",
  },
  "hoveredItem": null,
}
`;

exports[`Accepted ChartGroup:  should handle breadcrumb 2`] = `
Object {
  "active": null,
  "hoveredItem": null,
}
`;

exports[`Accepted ChartGroup:  should handle mouse in and out 1`] = `
Object {
  "active": null,
  "hoveredItem": Object {
    "count": 5,
    "dataSet": undefined,
    "index": 0,
    "name": "first attempt",
  },
}
`;

exports[`Accepted ChartGroup:  should handle mouse in and out 2`] = `
<PieChart.ActiveLabel
  name="first attempt"
  value="10%"
/>
`;

exports[`Accepted ChartGroup:  should handle mouse in and out 3`] = `
<PieChart.ActiveLabel
  name="Accepted Rate"
  value="62.5%"
/>
`;

exports[`Accepted ChartGroup:  should handle mouse in and out 4`] = `
Object {
  "active": null,
  "hoveredItem": null,
}
`;

exports[`Accepted ChartGroup:  should not do anything if clicking on an item without children 1`] = `
Object {
  "active": null,
  "hoveredItem": null,
}
`;

exports[`Accepted ChartGroup:  should render 1`] = `
<Panel
  sectioned={true}
  title="Accepted Rates"
>
  <Grid>
    <Grid.Column
      lg={5}
      xs={12}
    >
      <div>
        <LoadableComponent
          hoveredItem={null}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          primaryData={
            Array [
              Object {
                "count": 5,
                "fill": "hsl(85.80000000000001, 57.6%, 51%)",
                "name": "first attempt",
              },
              Object {
                "children": Array [
                  Object {
                    "count": 2,
                    "name": "2-4",
                  },
                  Object {
                    "count": 3,
                    "name": "5+",
                  },
                ],
                "count": 1,
                "fill": "hsl(65.80000000000001, 57.6%, 51%)",
                "name": "2 or more attempts",
              },
            ]
          }
        />
        <PieChart.ActiveLabel
          name="Accepted Rate"
          value="62.5%"
        />
      </div>
    </Grid.Column>
    <Grid.Column
      lg={7}
      xs={12}
    >
      <PieChart.Legend
        headerData={
          Array [
            Object {
              "count": 50,
              "name": "Accepted",
            },
            Object {
              "count": 80,
              "name": "Sent",
            },
          ]
        }
        hoveredItem={null}
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        primaryData={
          Array [
            Object {
              "count": 5,
              "fill": "hsl(85.80000000000001, 57.6%, 51%)",
              "name": "first attempt",
            },
            Object {
              "children": Array [
                Object {
                  "count": 2,
                  "name": "2-4",
                },
                Object {
                  "count": 3,
                  "name": "5+",
                },
              ],
              "count": 1,
              "fill": "hsl(65.80000000000001, 57.6%, 51%)",
              "name": "2 or more attempts",
            },
          ]
        }
      />
    </Grid.Column>
  </Grid>
</Panel>
`;

exports[`Accepted ChartGroup:  should render correctly after click 1`] = `
<PieChart.Legend
  headerData={
    Array [
      Object {
        "breadcrumb": true,
        "count": 50,
        "name": "Accepted",
        "onClick": [Function],
      },
      Object {
        "count": 80,
        "name": "Sent",
      },
      Object {
        "count": 1,
        "name": "2 or more attempts",
      },
    ]
  }
  hoveredItem={null}
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  primaryData={
    Array [
      Object {
        "count": 2,
        "fill": "hsl(85.80000000000001, 57.6%, 51%)",
        "name": "2-4",
      },
      Object {
        "count": 3,
        "fill": "hsl(65.80000000000001, 57.6%, 51%)",
        "name": "5+",
      },
    ]
  }
/>
`;

exports[`Accepted ChartGroup:  should render correctly after click 2`] = `
<LoadableComponent
  hoveredItem={null}
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  primaryData={
    Array [
      Object {
        "count": 2,
        "fill": "hsl(85.80000000000001, 57.6%, 51%)",
        "name": "2-4",
      },
      Object {
        "count": 3,
        "fill": "hsl(65.80000000000001, 57.6%, 51%)",
        "name": "5+",
      },
    ]
  }
/>
`;

exports[`Accepted ChartGroup:  should render label when hovering over a child 1`] = `
<PieChart.ActiveLabel
  name="2-4"
  value="4%"
/>
`;

exports[`Accepted ChartGroup:  should render loading 1`] = `
<Panel
  sectioned={true}
  title="Accepted Rates"
>
  <Loading />
</Panel>
`;
